sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -110,6 +110,27 @@ event submit_with_event_impl(const queue &Q, PropertiesT Props,
   return Q.submit_with_event<__SYCL_USE_FALLBACK_ASSERT>(
       Props, detail::type_erased_cgfo_ty{CGF}, nullptr, CodeLoc);
 }
+
+#ifdef __DPCPP_ENABLE_UNFINISHED_NO_CGH_SUBMIT
+template <typename KernelName, typename PropertiesT, typename KernelType,
+          int Dims>
+void submit_direct_impl(const queue &Q, PropertiesT Props, nd_range<Dims> Range,
+                        const KernelType &KernelFunc,
+                        const sycl::detail::code_location &CodeLoc) {
+  Q.submit_direct_without_event<KernelName, __SYCL_USE_FALLBACK_ASSERT,
```

Comment:
I don't think you need to handle the fallback assert here, it's being removed: https://github.com/intel/llvm/pull/18310